libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -860,8 +860,8 @@ string ABIFunctions::abiEncodingFunctionStruct(
 		for (auto const& member: _to.members(nullptr))
 		{
 			solAssert(member.type, "");
-			if (!member.type->canLiveOutsideStorage())
-				continue;
+			solAssert(member.type->nameable(), "");
```

Comment:
If we add the nameable ossert in `containsNestedMapping` then we actually don't need it here.